pkg/operator/ceph/config/conditions.go

```diff
@@ -128,6 +137,61 @@ func ConditionInitialize(context *clusterd.Context, namespace, name string) {
 		Reason:  "",
 		Message: "",
 	})
+	setCondition(context, namespace, name, cephv1.Condition{
+		Type:    cephv1.ConditionAvailable,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+}
+
+// GetOSDDeployments returns an array containing the names of all OSD
+func GetOSDDeployments(context *clusterd.Context, namespace, AppName string) ([]string, error) {
+	var osdIDArray []string
+	listOpts := metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", AppName)}
+	deployments, err := context.Clientset.AppsV1().Deployments(namespace).List(listOpts)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get exsting deployment %q", AppName)
+	}
+	for _, osdDeployments := range deployments.Items {
+		osdIDArray = append(osdIDArray, osdDeployments.Name)
+	}
+	return osdIDArray, nil
+}
+
+// ExpandConditionCheck returns true if given osdID is already in the osdIDArray
+func ExpandConditionCheck(osdIDArray []string, osdID int) bool {
+	for i := range osdIDArray {
+		if i == osdID {
+			return true
+		}
+	}
+	return false
+}
+
+// AvailableMapping check if Storage is available or not
+func AvailableMapping() (v1.ConditionStatus, string, string) {
+	if conditionMap[cephv1.ConditionAvailable] == v1.ConditionTrue {
+		return v1.ConditionTrue, "StorageAvailable", "Storage is available for cluster"
+	}
+	return v1.ConditionFalse, "StorageNotAvailable", "Storage is not available for cluster"
+}
+
+// RemoveOSDDeployments removes an OSD from the osdIDArray
+func RemoveOSDDeployments(outOSDid int, osdIDArray []string) []string {
+	var index int
+	for count, id := range osdIDArray {
```

Comment:
I have a feeling that using a map instead would make things easier here.

Reply:
I'll try that, thanks.